### ISSUE TITLE
no-unclosed-element-at-eof false-positives on the last element with parsers that never emit close tags (e.g. pug-parser)

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#4022]
## Summary

`no-unclosed-element-at-eof` reports a false positive on the last element of a document when the parser never records explicit close tags — e.g. `@markuplint/pug-parser`, since Pug syntax has no closing-tag construct at all.

## Reproduction

`markuplint.config.mjs`:

```js
export default {
	extends: ['markuplint:recommended-static-html'],
	parser: { '.pug$': '@markuplint/pug-parser' },
};
```

`sample.pug`:

```pug
doctype html
html(lang="ja")
    head
        title Sample
        meta(charset="UTF-8")
    body
        div Hello
```

```
$ markuplint --config ./markuplint.config.mjs sample.pug
<markuplint> error: The "div" element is not closed at the end of the file [normative] (html-standard/no-unclosed-element-at-eof) sample.pug:7:9
```

The last non-exempt, non-void element in the file is always flagged, regardless of its tag name (tested with `div`; `p`, `li`, `body`, `html` are unaffected only because they're in the rule's exemption list, not because they're actually closed).

## Root cause

- `no-unclosed-element-at-eof` (`packages/@markuplint/rules/src/no-unclosed-element-at-eof/index.ts:77-81`) reports any element where `el.closeTag == null`, unless it's void or in `EOF_EXEMPT_NAMES` (same file, lines 25-44 — the HTML Living Standard's optional-tag-omission list).
- `@markuplint/pug-parser` extends the base `Parser` with `endTagType: 'never'` (`packages/@markuplint/pug-parser/src/parser.ts:44-48`), because Pug's indentation-based syntax never has an explicit closing tag for *any* element.
- Consequently every element parsed from a `.pug` file has `closeTag == null`, and the last one in the source (that isn't in the exemption list) is reported as "not closed at the end of the file" — even though this is simply how the source language works, not a malformed document.

## Suggested fix

The rule's exemption logic assumes "no close tag" implies a possible HTML parse error. That assumption doesn't hold for a parser that structurally never emits close tags. Either:
- Skip this rule entirely for parsers declaring `endTagType: 'never'`, or
- Have such parsers surface that fact on the AST node so the rule can exclude it explicitly.

## Environment

- `markuplint@5.0.0-rc.6` / `@markuplint/pug-parser@5.0.0-rc.6` (also confirmed against the current `dev` branch HEAD)
